apiserversdk/proxy.go

```diff
@@ -84,3 +92,103 @@ func requireKubeRayService(handler http.Handler, k8sClient *kubernetes.Clientset
 		handler.ServeHTTP(w, r)
 	})
 }
+
+// retryRoundTripper is a custom implementation of http.RoundTripper that retries HTTP requests.
+// It verifies retryable HTTP status codes and retries using exponential backoff.
+type retryRoundTripper struct {
+	base    http.RoundTripper
+	retries int
+}
+
+func newRetryRoundTripper(base http.RoundTripper, retries int) http.RoundTripper {
```

Comment:
On testing, `RoundTripper` is an [interface](https://pkg.go.dev/net/http#RoundTripper), which means you can have your mock implementation to mimic error; I think we should be able to test retry and timeout with unit test?